modules/ppcp-settings/docs/merchant-onboarding-flow.md

```diff
@@ -0,0 +1,263 @@
+- **Determination:** `$container->get('wc-subscriptions.helper')->plugin_is_active()`
+
+**`should_skip_payment_methods`** - Skip payment methods selection screen
+- **Source:** Direct class existence check
+- **Logic:** `class_exists('\WC_Payments')`
+- **Purpose:** Skip payment methods screen when WooCommerce Payments plugin is active (avoid conflicts)
+
+**`can_use_fastlane`** - Fastlane/AXO checkout capability
+- **Source:** `axo.eligible` service from AXO module
+- **Dependencies:** Requires ACDC capability and specific country/merchant eligibility
+- **Purpose:** Determines if Advanced Checkout Options (one-click checkout) can be offered
+
+**`can_use_pay_later`** - Pay Later messaging availability
+- **Source:** `button.helper.messages-apply` service from Button module
+- **File:** `modules/ppcp-button/services.php` (MessagesApply class)
+- **Logic:** `MessagesApply::for_country()` - checks country against PayPal's Pay Later supported regions
+- **Countries:** Pay Later messaging supported in US, AU, DE, FR, GB, IT, ES and other select markets
+
+#### Country-Based Eligibility Services
+
+Multiple modules implement country-based eligibility patterns:
+
+```php
+// Common pattern across modules
+'module.helpers.applies' => function(ContainerInterface $container): ModuleApplies {
+    return new ModuleApplies(
+        $container->get('module.supported-countries'),  // Country whitelist
+        $container->get('api.shop.country')            // Current store country
+    );
+},
+
+// Eligibility check
+'module.eligible' => function(ContainerInterface $container): bool {
+    $applies = $container->get('module.helpers.applies');
+    return $applies->for_country() && $applies->for_merchant();
+},
+```
+
+#### Casual Selling Country List
+
+The casual selling feature (personal PayPal accounts) supported countries are defined in: `modules/ppcp-settings/services.php`
+
+### API Endpoint: `wc/v3/wc_paypal/onboarding`
+
+The central REST API endpoint manages all onboarding state and configuration:
+
+**File:** `modules/ppcp-settings/src/Endpoint/OnboardingRestEndpoint.php`
+
+#### Endpoints:
+- **GET** `/wp-json/wc/v3/wc_paypal/onboarding` - Retrieves current onboarding state and flags
+- **POST** `/wp-json/wc/v3/wc_paypal/onboarding` - Updates onboarding progress and settings
+
+#### Key Data Structures:
+
+```php
+// Onboarding state fields (OnboardingProfile)
+'completed'            => bool    // Onboarding process completion status
+'step'                 => int     // Current step index (0-based)
+'is_casual_seller'     => bool    // Personal vs business account selection
+'accept_card_payments' => bool    // Whether to enable card payment methods
+'products'             => array   // Selected product types ['virtual', 'physical', 'subscriptions']
+'gateways_synced'      => bool    // Payment gateway sync status
+'gateways_refreshed'   => bool    // Payment gateway refresh status
```

Comment:
Suggestion: Could we add those code comments to the actual `.php` file, e.g. to the `OnboardingProfile` and drop a link here? Those comments are very helpful, but developers usually do not see them in this MD file

https://github.com/woocommerce/woocommerce-paypal-payments/blob/060a86f5c9693872328e4fbb9f58718e8456d091/modules/ppcp-settings/src/Data/OnboardingProfile.php#L77-L88